pkg/kubelet/kubelet.go

```diff
@@ -2010,6 +2010,13 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 
 	// Wait for volumes to attach/mount
 	if err := kl.volumeManager.WaitForAttachAndMount(ctx, pod); err != nil {
+		var volumeAttachLimitErr *volumemanager.VolumeAttachLimitExceededError
+		if errors.As(err, &volumeAttachLimitErr) {
+			reasonStr := "VolumeAttachmentLimitExceeded"
+			kl.rejectPod(pod, reasonStr, "Node has insufficient volume attachment capacity")
+			recordAdmissionRejection(reasonStr)
```

Comment:
I would suggest we have another reason for admission failure metric - please update the `recordAdmissionRejection` to include an extra metric label value

Reply:
I think what @SergeyKanzhelev perhaps meant is, enum value in - https://github.com/AndrewSirenko/kubernetes/blob/vac-valid/pkg/kubelet/kubelet.go#L239 , so as metric gets emitted with proper reason, otherwise I think, it will simply show up as "Other" reason.

Reply:
@gnufied Yep, thats exactly what was implemented - see https://github.com/kubernetes/kubernetes/pull/132933/commits/e93a9edd23158114f3bcafdfa33c67c753aa5b71 (theres also a unit test for this in kubelet_test.go).